### ISSUE TITLE
Use project(sigstore-java) in sigstore-gradle-plugin tests

### ISSUE THREAD

**Issue by loosebazooka** [sigstore/sigstore-java#204]
Changing sigstore-java requires changes in sigstore-gradle-plugin, but then some tests go out of sync (as they depend on maven central) and then fail.

**Comment by vlsi** [comment]
Could you clarify? Which changes and which failures do you mean?

**Comment by loosebazooka** [comment]
If for example we change the name of `KeylessSigner.sign` to `KeylessSigner.sign2`. We have to change the code in `SignWorkAction.kt` to reflect this. But a test that applies the plugin like the gradle plugin test `SigstoreSignTest` will fail with:
```
Caused by: java.lang.NoSuchMethodError: 'dev.sigstore.KeylessSigningResult dev.sigstore.KeylessSigner.sign2(java.n
io.file.Path)'
```

**Comment by loosebazooka** [comment]
@vlsi, any updates on this, it's blocking some changes.

**Comment by loosebazooka** [comment]
I think is also blocking any fix for #212

**Comment by vlsi** [comment]
For backward-incompatible changes, we should probably use reflection. In other words, `SignWorkAction` should call `KeylessSigner` via reflection, so it could support multiple `KeylessSigner` versions.

Then, it looks like that the proper fix for #212  would to expose the set of trust roots to the users. In other words, there should be a plugin option (and the relevant sigstore-java option) so the clients could supply the root cert.
